hadoop-ozone/common/src/main/proto/OzoneManagerProtocol.proto

```diff
@@ -139,7 +140,8 @@ message OMRequest {
   optional MultipartUploadListPartsRequest  listMultipartUploadPartsRequest = 50;
 
   optional ServiceListRequest               serviceListRequest             = 51;
-  optional DBUpdatesRequest                  dbUpdatesRequest              = 53;
+  optional OMServerRolesRequest             omServerRolesRequest           = 52;
+  optional DBUpdatesRequest                 dbUpdatesRequest               = 53;
```

Comment:
So today when a client connects to the OM -- it connect reads to get ServiceList, without this API we cannot communicate to Ozone. So Why would this info be an independent API, since in my mind this info is needed by the clients to communicate to OM, since the leader needs to be discovered.

if that is true, then admin command can use the same API do discover the state right ? do we need a new API in the RPC layer? Perhaps I am missing something here, Thanks

Reply:
The client connects to an OM and submits a getServiceList request. If this OM is not the leader, then the client tries a different OM. This all happens through _OMFailoverProxyProvider_. 
Currently, the _getServiceList_ API is only used to discover the SCM address. It does not return the list of OMs in the cluster.

But based on which OM serves the getServiceList request, we can presume it to be the leader. We do not need a new request type. I will update the patch.